clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinator.java

```diff
@@ -209,9 +209,10 @@ public String protocolType() {
 
         List<String> topics = new ArrayList<>(joinedSubscription);
         for (ConsumerPartitionAssignor assignor : assignors) {
+            final List<TopicPartition> ownedPartitions = subscriptions.assignedPartitionsList();
             Subscription subscription = new Subscription(topics,
-                                                         assignor.subscriptionUserData(joinedSubscription),
-                                                         subscriptions.assignedPartitionsList());
+                                                         assignor.subscriptionUserData(joinedSubscription, ownedPartitions),
+                                                         ownedPartitions);
```

Comment:
Here we pass in the `ownedPartitions` that will be included in the Subscription so that the assignor can remove any that it knows are safe to reassign without first revoking (as is the case for non-running or restoring tasks, ie any whose partitions would remain paused)